### PR TITLE
Adds buildTarget/didChange notification to the build server protocol

### PR DESCRIPTION
**Pull request by rmaz**
This PR adds the `buildTarget/didChange` notification to the build server protocol, as well as a `BuildSystemDelegate` method that is called when it is received.

### CODE REVIEW

File: Sources/BuildServerProtocol/BuildTargets.swift [hunk 1]
```diff
@@ -207,3 +207,44 @@ public struct OutputsItem: Codable, Hashable {
...
+    self.changes = changes
+  }
+}
+
+public struct BuildTargetEvent: Codable, Hashable {
+  /// The identifier for the changed build target.
+  public var target: BuildTargetIdentifier
+
+  /// The kind of change for this build target.
+  public var kind: BuildTargetEventKind
```
Comment (benlangmuir) [inline Sources/BuildServerProtocol/BuildTargets.swift]: The spec says this should be optional (https://github.com/build-server-protocol/build-server-protocol/blob/master/docs/specification.md#build-target-changed-notification), although I'm not sure what the semantics of `nil` are here. Any thoughts?

File: Tests/SKCoreTests/BuildServerBuildSystemTests.swift [hunk 2]
```diff
@@ -155,19 +155,52 @@ final class BuildServerBuildSystemTests: XCTestCase {
     })
     XCTAssertEqual(XCTWaiter.wait(for: [expectation], timeout: 15), .completed)
   }
+
+  func testBuildTargetsChanged() {
+    let root = AbsolutePath(
+      inputsDirectory().appendingPathComponent(testDirectoryName, isDirectory: true).path)
+    let buildFolder = AbsolutePath(NSTemporaryDirectory())
+    let buildSystem = try? BuildServerBuildSystem(projectRoot: root, buildFolder: buildFolder)
```
Comment (benlangmuir) [inline Tests/SKCoreTests/BuildServerBuildSystemTests.swift]: Prefer `try` so that we will see the error if this fails.

**Comment by benlangmuir** [comment]
@swift-ci please test

File: Sources/BuildServerProtocol/BuildTargets.swift (hunk 1 above)
Comment (rmaz) [reply Sources/BuildServerProtocol/BuildTargets.swift]: Yes, that is rather confusing. This value is nice as it allows for not invalidating the build target graph when a new target is created or deleted, you can just request the updated sources and outputs for the changed module. I suppose if not set the language server has to just refresh everything. I'll update this to be optional.

**Comment by benlangmuir** [comment]
@swift-ci please test

**Comment by benlangmuir** [comment]
@swift-ci please test

**Comment by benlangmuir** [comment]
@swift-ci please test

**Pull request merged**